vendor/tracing/zero/destinations.ml

```diff
@@ -32,7 +31,72 @@ let direct_file_destination ?(buffer_size = 4096 * 16) ~filename () =
   (module Dest : Writer_intf.Destination)
 ;;
 
-let file_destination ~filename () = direct_file_destination ~filename ()
+let compressed_file_destination ?(buffer_size = 64 * 1024) ~filename () =
+  let buf = Iobuf.create ~len:buffer_size in
+  let compressed_size = buffer_size * 2 in
+  (* N.B. The size is 2x the buffer because, due to e.g. headers,
+     Zstandard will sometimes write slightly more data than the
+     original buffer had. *)
+  let compression_level = 5 in
+  let compressed_buf = Iobuf.create ~len:compressed_size in
+  let file = Core_unix.openfile ~mode:[ O_CREAT; O_TRUNC; O_CLOEXEC; O_RDWR ] filename in
+  let written = ref 0 in
+  let compression_context = Zstandard.Compression_context.create () in
+  let flush () =
```

Comment:
Does this compress the full trace file at once? Would it be easy to change this to stream the file out, instead? I worry that this may cause users with relatively large trace files to OOM for no good reason. In particular, those users are the most likely to care about compressing their trace files in the first place.